File: guides/micronaut-email/java/src/main/java/example/micronaut/MailController.java
package example.micronaut;

import io.micronaut.http.HttpResponse;
import io.micronaut.http.annotation.Body;
import io.micronaut.http.annotation.Controller;
import io.micronaut.http.annotation.Post;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import javax.validation.Valid;

@Controller("/mail") // <1>
public class MailController {

    private final EmailService emailService;

    public MailController(EmailService emailService) { // <2>
        this.emailService = emailService;
    }

    @Post("/send") // <3>
    public HttpResponse<?> send(@Body @Valid EmailCmd cmd) { // <4>
        emailService.send(cmd);
        return HttpResponse.ok();  // <5>
    }
}

File: guides/micronaut-email/kotlin/src/main/kotlin/example/micronaut/MailController.kt
package example.micronaut

import io.micronaut.http.HttpResponse
import io.micronaut.http.annotation.Body
import io.micronaut.http.annotation.Controller
import io.micronaut.http.annotation.Post
import javax.validation.Valid

@Controller("/mail") // <1>
open class MailController(private val emailService: EmailService) {  // <2>

    @Post("/send") // <3>
    open fun send(@Body @Valid cmd: EmailCmd): HttpResponse<*> {  // <4>
        emailService.send(cmd)
        return HttpResponse.ok<Any>() // <5>
    }
}
